src/music/player.ts

```diff
@@ -0,0 +1,162 @@
+import {
+  AudioPlayer,
+  AudioPlayerStatus,
+  createAudioPlayer,
+  createAudioResource,
+  joinVoiceChannel,
+  VoiceConnection,
+} from "@discordjs/voice";
+import { Guild, GuildMember } from "discord.js";
+import ytdl from "@distube/ytdl-core"; // ✅ Use maintained version
+import type { QueueItem } from "./queue.ts"; // ✅ Import as type
+import { MusicQueue } from "./queue.ts";
+
+export class MusicPlayer {
+  private players = new Map<string, AudioPlayer>();
+  private connections = new Map<string, VoiceConnection>();
+  private queues = new MusicQueue();
+
+  async play(
+    guild: Guild,
+    member: GuildMember,
+    query: string,
+  ): Promise<string> {
```

Comment:
You don't need the explicit return type here and in other places `Promise<string>`